Sources/Commands/Error.swift

```diff
@@ -80,7 +80,7 @@ func print(diagnostic: Diagnostic, stdoutStream: OutputByteStream) {
         break
     }
 
-    writer.write(diagnostic.description)
+    writer.write(diagnostic.description.replacingOccurrences(of: "\r", with: ""))
```

Comment:
It would be a bit cleaner to avoiding adding `\r` in the first place vs patching it here. Also note that `replacingOccurrences(of:with:)` will bridge to NSString on macOS which might result in some perf hit.

Reply:
The addition of `\r` is not under our control.  That is the line ending - line endings are always `\r\n` on Windows.  So, you will have to strip them somewhere.

Reply:
Oh you mean a new line is automatically converted into `\r\n`? I think we should at least put this under `#if os(windows)` if it's unavoidable 😕

Reply:
No, a newline is not converted into `\r\n`.  A line is ended by `\r\n`.  When the compiler prints a diagnostic that is read, it will have the `\r\n` ending.  I think that we should actively avoid any `#if os(Windows)` if the path can be supported on all platforms.

Reply:
So is this only a problem for compiler diagnostics? IIRC diagnostics from manifest loader are the only compiler diagnostics that go through diagnostics engine so maybe we can patch it there. I agree that we should avoid #if where we can but this seems like a completely unnecessary operation on other platforms.

Reply:
It may be unnecessary, but it doesn't hurt :)  No, the diagnostics that this impacts are not the manifest one, but the target libraries.  This change actually fixes the terminal corruption that I was seeing on Windows due to the line ending mixups not terminating the color'ed output and fixes the extraneous empty lines that I was getting everywhere.  If there is another place that is better for the diagnostics from the swift compiler and clang, then we should move it there, but this seemed to be the one funnel point for all of them.

Reply:
Ugh, so, this is still not 100% correct:

**Actual:**
```
:158:11: warning: logical not is only applied to the left hand side of this comparison [-Wlogical-not-parentheses]
      if (!S_ISDIR(statbuf.st_mode)) {
          ^~~~~~~~~~~~~~~~~~~~~~~~~
:158:11: note: add parentheses after the '!' to evaluate the comparison first
      if (!S_ISDIR(statbuf.st_mode)) {
          ^
           (                       )
S:\b\swift-package-manager\.build\checkouts\swift-llbuild\lib\Basic\FileSystem.cpp:158:11: note: add parentheses around left hand side expression to silence this warning
:158:12: warning: result of comparison of constant 16384 with expression of type 'bool' is always false [-Wtautological-constant-out-of-range-compare]
      if (!S_ISDIR(statbuf.st_mode)) {
          ~^~~~~~~~~~~~~~~~~~~~~~~~
: expanded from macro 'S_ISDIR'

                                         ^  ~~~~~~~
 warnings generated.
```

**Expected:**
```
S:\b\swift-package-manager\.build\checkouts\swift-llbuild\lib\Basic\FileSystem.cpp:158:11: warning: logical not is only applied to the left hand side of this comparison [-Wlogical-not-parentheses]

      if (!S_ISDIR(statbuf.st_mode)) {

          ^~~~~~~~~~~~~~~~~~~~~~~~~

S:\b\swift-package-manager\.build\checkouts\swift-llbuild\lib\Basic\FileSystem.cpp:158:11: note: add parentheses after the '!' to evaluate the comparison first

      if (!S_ISDIR(statbuf.st_mode)) {

          ^

           (                       )

S:\b\swift-package-manager\.build\checkouts\swift-llbuild\lib\Basic\FileSystem.cpp:158:11: note: add parentheses around left hand side expression to silence this warning

S:\b\swift-package-manager\.build\checkouts\swift-llbuild\lib\Basic\FileSystem.cpp:158:12: warning: result of comparison of constant 16384 with expression of type 'bool' is always false [-Wtautological-constant-out-of-range-compare]

      if (!S_ISDIR(statbuf.st_mode)) {

          ~^~~~~~~~~~~~~~~~~~~~~~~~

S:\b\swift-package-manager\.build\checkouts\swift-llbuild\lib\Basic\include\llbuild/Basic/Stat.h:42:42: note: expanded from macro 'S_ISDIR'

#define S_ISDIR(mode) ((mode) & _S_IFMT) == S_IFDIR

                                         ^  ~~~~~~~

2 warnings generated.
```